ShellPkg/Library/UefiShellDebug1CommandsLib/LoadPciRom.c

```diff
@@ -333,6 +338,10 @@ LoadEfiDriversFromRomImage (
           //
           UnicodeSPrint (RomFileName, sizeof (RomFileName), L"%s[%d]", FileName, ImageIndex);
           FilePath = FileDevicePath (NULL, RomFileName);
+          if (FilePath == NULL) {
+            ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_LOADPCIROM_LOAD_FAIL), gShellDebug1HiiHandle, L"loadpcirom", FileName, ImageIndex);

```

Comment:
`DecompressedImageBuffer` should be freed

Reply:
Added.